src/test/regress/expected/with_clause.out

```diff
@@ -2180,3 +2180,174 @@ WITH RECURSIVE r1 AS (
+                                 Hash Key: i.i
+                                 ->  Function Scan on generate_series i
+ Optimizer: Postgres query optimizer
+(9 rows)
+
+with cte as (
+    insert into with_dml select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
+ count 
+-------
+     5
+(1 row)
+
+explain (costs off)
+with cte as (
+    insert into with_dml_dr select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
+                          QUERY PLAN                           
+---------------------------------------------------------------
+ Aggregate
+   ->  Subquery Scan on cte
+         ->  Explicit Gather Motion 3:1  (slice1; segments: 3)
+               ->  Insert on with_dml_dr
+                     ->  Function Scan on generate_series i
+ Optimizer: Postgres query optimizer
+(6 rows)
+
+with cte as (
+    insert into with_dml_dr select i, i * 100 from generate_series(1,5) i
+    returning i
+) select count(*) from cte;
+ count 
+-------
+     5
+(1 row)
+
+-- Test select over update queries. Check no qual pushed down to update from
+-- select, because it's incorrect to push it down in such way. Queries should
+-- be correctly executed without any rewriting error, e.g. rewriteManip.c:1409.
+explain (costs off)
+with cte as (
+    update with_dml set j = j + 1 where i <= 5 returning j
+) select count(*) from cte where j > 100;
+                     QUERY PLAN                     
+----------------------------------------------------
+ Aggregate
+   ->  Gather Motion 3:1  (slice1; segments: 3)
+         ->  Aggregate
+               ->  Subquery Scan on cte
+                     Filter: (cte.j > 100)
+                     ->  Update on with_dml
+                           ->  Seq Scan on with_dml
+                                 Filter: (i <= 5)
+ Optimizer: Postgres query optimizer
+(9 rows)
+
+with cte as (
+    update with_dml set j = j + 1 where i <= 5 returning j
+) select count(*) from cte where j > 100;
+ count 
+-------
+     5
+(1 row)
+
+explain (costs off)
+with cte as (
+    update with_dml_dr set j = j + 1 where i <= 5 returning j
+) select count(*) from cte where j > 100;
+                          QUERY PLAN                           
+---------------------------------------------------------------
+ Aggregate
+   ->  Subquery Scan on cte
+         Filter: (cte.j > 100)
+         ->  Explicit Gather Motion 3:1  (slice1; segments: 3)
+               ->  Update on with_dml_dr
+                     ->  Seq Scan on with_dml_dr
+                           Filter: (i <= 5)
+ Optimizer: Postgres query optimizer
+(8 rows)
+
+with cte as (
+    update with_dml_dr set j = j + 1 where i <= 5 returning j
+) select count(*) from cte where j > 100;
+ count 
+-------
+     5
+(1 row)
+
+-- Test select over delete queries. Test joining, which caused fatal error
+-- (cdbpath.c:1105) due to incompatibility of using CdbLocusType_Replicated
```

Comment:
Please specify what is the fatal error - i.e. failed assertion at [cdbpath.c:1105](https://github.com/arenadata/gpdb/blob/2aea19802afbac1d0fa2f4b7c7d7eec5de9e3eaa/src/backend/cdb/cdbpath.c#L1105) as line number can change later:

` Assert(root->upd_del_replicated_table > 0);`